### PR TITLE
handle submodule in extension template

### PR DESCRIPTION
**Pull request by ManishPrajapati-GoFynd**

### CODE REVIEW

File: src/lib/Extension.ts [hunk 1]
```diff
@@ -49,23 +51,57 @@ export default class Extension {
...
             await execa(
                 'git',
                 ['remote', 'add', 'origin', answers.project_url],
-                { cwd: targetDirectory },
+                { cwd: tempDirectory },
             );
-            await execa('git', ['pull', 'origin', 'main:main'], {
-                cwd: targetDirectory,
+            
+            await execa('git', ['pull', '--recurse-submodules', 'origin', 'main:main'], {
```
Comment (vivek-gofynd) [inline src/lib/Extension.ts]: Keep this branch name as configurable @manishprajapati52

File: src/lib/Extension.ts [hunk 2]
```diff
@@ -49,23 +51,57 @@ export default class Extension {
...
+            
+            rimraf.sync(`${tempDirectory}/.git`); // unmark as git repo
+            rimraf.sync(`${tempDirectory}/.gitmodules`); // Remove the .gitmodules file
+            
+            const submoduleGitPath = `${tempDirectory}/${submodulePath}/.git`
+            submodulePath && Debug(`Deleting ${submoduleGitPath}`)
+            submodulePath && rimraf.sync(submoduleGitPath); // unmark as git repo from submodules
+            
+            // move project from temporary directory to main directory
+            const files = glob.sync(path.join(tempDirectory, '{*,.*}'));
```
Comment (vivek-gofynd) [inline src/lib/Extension.ts]: Create common helper function with doc string comments and details of parameters @manishprajapati52

**Review by vivek-gofynd — CHANGES_REQUESTED** [review]

File: src/lib/Extension.ts (hunk 1 above)
Comment (ManishPrajapati-GoFynd) [reply src/lib/Extension.ts]: @vivek-gofynd 
Didn't understand this.
Configurable as in? 
Each project can have different branch?

File: src/lib/Extension.ts [hunk 3]
```diff
@@ -49,23 +51,57 @@ export default class Extension {
         targetDirectory: string,
         answers: Object,
     ) {
+        const tempDirectory = targetDirectory + '/temp';
```
Comment (vivek-gofynd) [inline src/lib/Extension.ts]: keep this as constant and keep it in constant file and manage it centrally. @manishprajapati52

File: src/lib/Extension.ts (hunk 1 above)
Comment (ManishPrajapati-GoFynd) [reply src/lib/Extension.ts]: Done

File: src/lib/Extension.ts (hunk 2 above)
Comment (ManishPrajapati-GoFynd) [reply src/lib/Extension.ts]: Done

**Review by vivek-gofynd — CHANGES_REQUESTED** [review]

**Comment by ManishPrajapati-GoFynd** [comment]
@vivek-gofynd 
We can close this as @bhargavprajapatiFynd has taken pull of this branch in his feature branch.

**Pull request closed without merging**